src/main/java/seedu/address/model/Alfred.java

```diff
@@ -0,0 +1,377 @@
+package seedu.address.model;
+
+import java.io.IOException;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import java.util.Optional;
+import java.util.logging.Logger;
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.exceptions.AlfredException;
+import seedu.address.commons.exceptions.AlfredModelException;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.model.entity.Id;
+import seedu.address.model.entity.Mentor;
+import seedu.address.model.entity.Participant;
+import seedu.address.model.entity.PrefixType;
+import seedu.address.model.entity.Team;
+import seedu.address.model.entitylist.MentorList;
+import seedu.address.model.entitylist.ParticipantList;
+import seedu.address.model.entitylist.ReadOnlyEntityList;
+import seedu.address.model.entitylist.TeamList;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.UniquePersonList;
+import seedu.address.storage.AlfredStorage;
+
+/**
+ * Wraps all data at the alfred level
+ * Duplicates are not allowed (by .isSamePerson comparison)
+ */
+public class Alfred implements ReadOnlyAlfred {
+    private static final Logger logger = LogsCenter.getLogger(Alfred.class);
+
+    private final ParticipantList participantList;
+    private final MentorList mentorList;
+    private final TeamList teamList;
+
+
+    public Alfred() {
+        participantList = new ParticipantList();
+        teamList = new TeamList();
+        mentorList = new MentorList();
+    }
+
+    /**
+     * Creates an AddressBook using the Persons in the {@code toBeCopied}
+     */
+    public Alfred(AlfredStorage alfredStorage, ReadOnlyUserPrefs userPrefs) throws AlfredException {
+
+        Optional<ParticipantList> optionalParticipantList;
+        Optional<MentorList> optionalMentorList;
+        Optional<TeamList> optionalTeamList;
```

Comment:
Could we do this for the initalization instead of defining an optionalXList variable
````
try {
    this.participantList = alfredStorage.readParticipantList().orElse(new ParticipantList());
    ....
} catch (IOException e) {
    // Set all the entitylists to default value
    this.participantList = new ParticipantList();
    ....
}
````